### PR TITLE
Refactor dependency injection usage in entity framework

### PR DESCRIPTION
**Pull request by pakrym**
@davidfowl

### CODE REVIEW

File: src/EntityFramework.Commands/Extensions/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -8,23 +8,20 @@
...
-                foreach (var service in runtimeServices.Services)
-                {
-                    services.AddTransient(service, _ => dnxServices.GetService(service));
-                }
-            }
+            services.TryAdd(ServiceDescriptor.Instance(PlatformServices.Default.Application));
+            services.TryAdd(ServiceDescriptor.Instance(PlatformServices.Default.Runtime));
+            services.TryAdd(ServiceDescriptor.Instance(PlatformServices.Default.AssemblyLoadContextAccessor));
+            services.TryAdd(ServiceDescriptor.Instance(PlatformServices.Default.AssemblyLoaderContainer));
+            services.TryAdd(ServiceDescriptor.Instance(PlatformServices.Default.LibraryManager));
```
Comment (davidfowl) [inline src/EntityFramework.Commands/Extensions/ServiceCollectionExtensions.cs]: Add compilation services

**Comment by natemcmaster** [comment]
Looks good, but before signing off, let's verify it doesn't break any of the scenarios in https://github.com/aspnet/EntityFramework/issues/2294. By inspection, I'm not sure they will all work.

**Comment by natemcmaster** [comment]
:shipit:

File: src/EntityFramework.Commands/Commands/Program.cs [hunk 2]
```diff
@@ -23,26 +23,17 @@ namespace Microsoft.Data.Entity.Commands
...
-            _dnxServices = dnxServices;
+            _useConsoleColors = PlatformServices.Default.Runtime.OperatingSystem == "Windows";
         }
 
-        public virtual int Main([NotNull] string[] args)
+        public static int Main([NotNull] string[] args)
         {
             Check.NotNull(args, nameof(args));
 
-            Console.CancelKeyPress += (_, __) => _applicationShutdown.RequestShutdown();
```
Comment (bricelam) [inline src/EntityFramework.Commands/Commands/Program.cs]: Wait, we still need to cancel the async call in this case

**Comment by bricelam** [comment]
Was this supposed to go into RC1? I thought this work wasn't happening until RC2...

File: src/EntityFramework.Commands/Commands/Program.cs (hunk 2 above)
Comment (bricelam) [inline src/EntityFramework.Commands/Commands/Program.cs]: Fixed in #3567

**Pull request merged**